packages/eslint-plugin/lib/config/esnext.js

```diff
@@ -5,11 +5,11 @@ module.exports = {
   parser: 'babel-eslint',
 
   env: {
-    es6: true,
+    es2021: true,
   },
 
   parserOptions: {
-    ecmaVersion: 2018,
+    ecmaVersion: 'latest',
```

Comment:
```suggestion
    ecmaVersion: '2021',
```

`latest` doesn't seem to be supported in the parsers that we currently use. Let's keep this as `2021`, and look at using `latest` once we're using parsers that support it.

- We use `@typescript-eslint/parser@^4.28.2` however support for ecmaVersion 'latest' was added in [v4.32.0](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.32.0)
- We use the long-deprecated `babel-eslint` however  support for ecmaVersion 'latest' was added in it's successor `@babel/eslint-parser` in [v7.15.4](https://github.com/babel/babel/pull/13638)